Cheetah/Tests/Unicode.py

```diff
@@ -13,33 +13,25 @@
 
 class CommandLineTest(unittest.TestCase):
     def createAndCompile(self, source):
-        fd, sourcefile = tempfile.mkstemp()
-        os.close(fd)
-        os.remove(sourcefile)
-        sourcefile = sourcefile.replace('-', '_')
-
-        if PY2:
-            fd = open('%s.tmpl' % sourcefile, 'w')
-        else:
-            fd = open('%s.tmpl' % sourcefile, 'w', encoding='utf-8')
-        fd.write(source)
-        fd.close()
-
-        wrap = CheetahWrapper.CheetahWrapper()
-        wrap.main(['cheetah', 'compile',
-                   '--encoding=utf-8', '--settings=encoding="utf-8"',
-                   '--quiet', '--nobackup', sourcefile])
-        module_name = os.path.split(sourcefile)[1]
-        module = load_module_from_file(
-            module_name, module_name, sourcefile + '.py')
-        template = getattr(module, module_name)
-        os.remove('%s.tmpl' % sourcefile)
-        for sourcefile_py in glob('%s.py*' % sourcefile):  # *.py[co]
-            os.remove(sourcefile_py)
-        __pycache__ = os.path.join(os.path.dirname(sourcefile), '__pycache__')
-        if os.path.exists(__pycache__):  # PY3
-            rmtree(__pycache__)
-        return template
+        with tempfile.TemporaryDirectory() as tempdir:
```

Comment:
`TemporaryDirectory` was added in `tempfile` in [Python 3.2](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) so the code doesn't work with Python 2.7 which is still a requirement.

Reply:
I can accept a pull request that doesn't immediately work with Python 2.7 but could be fixed. Unfortunately this one cannot be fixed.

Reply:
IMO the simplest and the smallest fix would be to replace the line
```python
sourcefile = sourcefile.replace('-', '_')
```
with 3 lines: `os.path.split` `sourcefile` into dirname and basename, fix basename and join dirname and basename back to the full path.

Reply:
This won't change the fact that the logic is entirely insecure, and you're not using `mkstemp()` correctly at all.

Reply:
> This won't change the fact that the logic is entirely insecure, and you're not using `mkstemp()` correctly at all.

These are just tests, nobody runs them in production, so no, there are no security issues. You're free to fix the problem in any way (use `mkdtemp` if you wish but don't forget to clean up), the primary requirement is — the entire test suite must pass; currently it [doesn't](https://github.com/CheetahTemplate3/cheetah3/actions/runs/4582329003) and cannot be fixed.

Reply:
Ok, I've used the same solution as in `Cheetah/Tests/Template.py` and I've fixed flake8 failures.